pkg/webhook/validation/edgeconnect/service_account.go

```diff
@@ -0,0 +1,19 @@
+package edgeconnect
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1alpha1/edgeconnect"
+	"golang.org/x/net/context"
+)
+
+const (
+	errorInvalidServiceName = `The EdgeConnect's specification has an invalid serviceAccountName.
+`
+)
+
+func isInvalidServiceName(_ context.Context, _ *edgeconnectValidator, edgeConnectCR *edgeconnect.EdgeConnect) string {
+	if edgeConnectCR.Spec.ServiceAccountName == "" {
+		return errorInvalidServiceName
+	}
+
+	return ""
```

Comment:
maybe dumb questions:
- Do you always need a `ServiceAccount` or only when `KubernetesApiAutomation` is used?

If so:
- We shouldn't always error for an empty `ServiceAccount`
- Why isn't the `ServiceAccountName` under the `KubernetesApiAutomation` section, will it be used for something else in the future?

Reply:
fixed [commit](https://github.com/Dynatrace/dynatrace-operator/pull/2977/commits/eb8a00a44ca0b1f7830ee270730082997d74f2aa)

Reply:
but then you have to have a `*string` for it in the CR, no?

because you want to differentiate between:
1. "not  set" == `nil`
2. "set to nothing" == `""`
3. "set to something" == `myServiceAccount`

A normal `string` can only have 2 states, so it wont work for:
1. "not  set" == `""` ⚠️ 
2. "set to nothing" == `""` ⚠️ 
3. "set to something" == `myServiceAccount`

Reply:
`serviceAccountName` is NOT optional. We don't need a pointer and `nil` value. It always has a value - either the default one or provided by an user. The problem is one can use empty string and apply such a CR because it is correct manifest. However from our/k8s point of view `""` value has no meaning so has to be immediately rejected.